lib/compiler/src/beam_ssa_type.erl

```diff
@@ -2148,6 +2148,8 @@ type(update_tuple, [Src | Updates], _Anno, Ts, _Ds) ->
 type(wait_timeout, [#b_literal{val=infinity}], _Anno, _Ts, _Ds) ->
     %% Waits forever, never reaching the 'after' block.
     beam_types:make_atom(false);
+type(bs_init_writable, [_Size], _, _, _) ->
+    #t_bitstring{};
```

Comment:
The type will be better if you set the type in the same way as for an empty binary:

```suggestion
    beam_types:make_type_from_value(<<>>);
```

With this change, the compiler will eliminate the `is_binary/1` test in the following example:

```
foo(L) ->
    Bs = << <<C:8>> || C <- L >>,
    true = is_binary(Bs),                       %Assertion.
    Bs.
```